src/ppom-controller.ts

```diff
@@ -992,8 +1030,12 @@ export class PPOMController extends BaseControllerV2<
    */
   async #jsonRpcRequest(
     method: string,
-    params: Record<string, unknown>,
-  ): Promise<any> {
+    params: JsonRpcParams,
+  ): Promise<
+    | JsonRpcSuccess<Json>
+    | (Omit<JsonRpcFailure, 'error'> & { error: unknown })
```

Comment:
I used the `JsonRpcResponse<Json>` type here instead of `unknown`, even though this technically narrows the response type in the `sendAsync` callback compared to `any` (see line 1059).

This is because `JsonRpcEngine` is where the callback passed into `sendAsync` actually gets invoked, and that class consistently uses the `JsonRpcResponse` type for responses.

The `error` type in `JsonRpcFailure` is widened to `unknown` since that's the only type accepted by `sendAsync`. See https://github.com/MetaMask/ppom-validator/pull/89/files#r1470601394 for a discussion on this point. 

Note: `sendAsync` is set to be deprecated in favor of `request` in a new EIP1193-compliant provider class.